lib/membrane/core/parent/structure_parser.ex

```
@@ -0,0 +1,77 @@
+defmodule Membrane.Core.Parent.StructureParser do
+  @moduledoc false
+  use Bunch
+
+  alias Membrane.Core.Parent.Link
+  alias Membrane.Core.Parent.Link.Endpoint
+  alias Membrane.{ChildrenSpec, Element, Pad, ParentError}
+
+  @type raw_link_t :: %Link{from: raw_endpoint_t(), to: raw_endpoint_t()}
+
+  @type raw_endpoint_t :: %Endpoint{
+          child: Element.name_t() | {Membrane.Bin, :itself},
+          pad_spec: Pad.name_t() | Pad.ref_t(),
+          pad_ref: Pad.ref_t() | nil,
+          pid: pid() | nil,
+          pad_props: map()
+        }
+
+  @spec parse(ChildrenSpec.structure_spec_t()) ::
+          {[raw_link_t], [ChildrenSpec.child_spec_extended_t()]} | no_return
+  def parse(structure) when is_list(structure) do
+    {links, children} =
+      structure
+      |> List.flatten()
+      |> Enum.map(fn
+        %ChildrenSpec.LinkBuilder{links: links, children: children, status: :done} ->
+          {Enum.reverse(links), Enum.reverse(children)}
+
+        %ChildrenSpec.LinkBuilder{links: [%{from: from} | _]} = builder ->
+          if length(builder.children) == 1 do
+            {[], builder.children}
+          else
+            raise ParentError,
+                  "Invalid link specification: link from #{inspect(from)} lacks its destination."
```

We should raise a better error when the status is just `from`, that is when someone just did `get_child` and didn't link it to anything. It's now not that obvious that it was supposed to be a start of a link